test/linalg/symmetric.jl

```diff
@@ -42,206 +41,229 @@ end
+            if eltya != BigFloat
+                @testset "trace" begin
+                    @test trace(asym) == trace(Hermitian(asym))
+                end
+                @testset "cond" begin
+                    @test cond(Hermitian(asym)) ≈ cond(asym)
+                end
+                
+                @testset "inversion" begin
+                    @test inv(Hermitian(asym)) ≈ inv(asym)
+                    if eltya <: Real && eltya != Int
+                        @test inv(Symmetric(asym)) ≈ inv(asym)
+                        @test inv(Hermitian(a)) ≈ inv(full(Hermitian(a)))
+                        @test inv(Symmetric(a)) ≈ inv(full(Symmetric(a)))
+                    end
+                end
+
+                @testset "symmetric eigendecomposition" begin
+                    d, v = eig(asym)
+                    @test asym*v[:,1] ≈ d[1]*v[:,1]
+                    @test v*Diagonal(d)*v' ≈ asym
+                    @test isequal(eigvals(asym[1]), eigvals(asym[1:1,1:1]))
+                    @test abs.(eigfact(Hermitian(asym), 1:2)[:vectors]'v[:,1:2]) ≈ eye(eltya, 2)
+                    eig(Hermitian(asym), 1:2) # same result, but checks that method works
+                    @test abs.(eigfact(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2)[:vectors]'v[:,1:2]) ≈ eye(eltya, 2)
+                    eig(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2) # same result, but checks that method works
+                    @test eigvals(Hermitian(asym), 1:2) ≈ d[1:2]
+                    @test eigvals(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2) ≈ d[1:2]
+                    @test full(eigfact(asym)) ≈ asym
+                    @test eigvecs(Hermitian(asym)) ≈ eigvecs(asym)
+
+                    # relation to svdvals
+                    @test sum(sort(abs.(eigvals(Hermitian(asym))))) == sum(sort(svdvals(Hermitian(asym))))
+                end
+                
+                @testset "det" begin
+                    @test det(asym) ≈ det(Hermitian(asym, :U))
+                    @test det(asym) ≈ det(Hermitian(asym, :L))
+                    if eltya <: Real
+                        @test det(asym) ≈ det(Symmetric(asym, :U))
+                        @test det(asym) ≈ det(Symmetric(asym, :L))
+                    end
+                    @test det(a + a.') ≈ det(Symmetric(a + a.', :U))
+                    @test det(a + a.') ≈ det(Symmetric(a + a.', :L))
+                end
+
+                @testset "rank" begin
+                    let A = a[:,1:5]*a[:,1:5]'
+                        # Make sure A is Hermitian even in the present of rounding error
+                        # xianyi/OpenBLAS#729
+                        A = (A' + A) / 2
+                        @test rank(A) == rank(Hermitian(A))
+                    end
+                end
+
+                @testset "pow" begin
+                    if eltya <: Real && eltya != Int
+                        @test (a*a')^2 ≈ Array(Symmetric(a*a')^2)
```

Comment:
Perhaps introduce a variable for `a*a'` at the top of this testset to save some CI time?